core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/adaptive/AdaptiveDeltaIntegralCodec.java

```diff
@@ -272,5 +295,169 @@ public double decodeDouble(double value) {
+          for (int i = 0; i < pageSize; i++) {
+            vector.putLong(i, (max - shortIntData[i]));
+          }
+        }  else if (dataType == DataTypes.TIMESTAMP) {
+          for (int i = 0; i < pageSize; i++) {
+            vector.putLong(i, (max - shortIntData[i]) * 1000);
+          }
+        } else if (DataTypes.isDecimal(dataType)) {
+          DecimalConverterFactory.DecimalConverter decimalConverter = vectorInfo.decimalConverter;
+          int precision = vectorInfo.measure.getMeasure().getPrecision();
+          for (int i = 0; i < pageSize; i++) {
+            BigDecimal decimal = decimalConverter.getDecimal(max - shortIntData[i]);
+            vector.putDecimal(i, decimal, precision);
+          }
+        } else {
+          for (int i = 0; i < pageSize; i++) {
+            vector.putDouble(i, (max - shortIntData[i]));
+          }
+        }
+      } else if (type == DataTypes.INT) {
+        int[] intData = columnPage.getIntData();
+        if (dataType == DataTypes.INT) {
+          for (int i = 0; i < pageSize; i++) {
+            vector.putInt(i, (int) (max - intData[i]));
+          }
+        } else if (dataType == DataTypes.LONG) {
+          for (int i = 0; i < pageSize; i++) {
+            vector.putLong(i, (max - intData[i]));
+          }
+        } else if (dataType == DataTypes.TIMESTAMP) {
+          for (int i = 0; i < pageSize; i++) {
+            vector.putLong(i, (max - intData[i]) * 1000);
+          }
+        } else if (DataTypes.isDecimal(dataType)) {
+          DecimalConverterFactory.DecimalConverter decimalConverter = vectorInfo.decimalConverter;
+          int precision = vectorInfo.measure.getMeasure().getPrecision();
+          for (int i = 0; i < pageSize; i++) {
+            BigDecimal decimal = decimalConverter.getDecimal(max - intData[i]);
+            vector.putDecimal(i, decimal, precision);
+          }
+        } else {
+          for (int i = 0; i < pageSize; i++) {
+            vector.putDouble(i, (max - intData[i]));
+          }
+        }
+      } else if (type == DataTypes.LONG) {
+        long[] longData = columnPage.getLongData();
+        if (dataType == DataTypes.LONG) {
+          for (int i = 0; i < pageSize; i++) {
+            vector.putLong(i, (max - longData[i]));
+          }
+        } else if (dataType == DataTypes.TIMESTAMP) {
+          for (int i = 0; i < pageSize; i++) {
+            vector.putLong(i, (max - longData[i]) * 1000);
+          }
+        } else if (DataTypes.isDecimal(dataType)) {
+          DecimalConverterFactory.DecimalConverter decimalConverter = vectorInfo.decimalConverter;
+          int precision = vectorInfo.measure.getMeasure().getPrecision();
+          for (int i = 0; i < pageSize; i++) {
+            BigDecimal decimal = decimalConverter.getDecimal(max - longData[i]);
+            vector.putDecimal(i, decimal, precision);
+          }
+        }
+      } else {
+        throw new RuntimeException("internal error: " + this.toString());
```

Comment:
required to optimize this exception message for better understanding

Reply:
ok